arclightning_backend/src/router.rs

```diff
@@ -143,4 +143,25 @@ impl Router {
             _ => self.invalid_endpoint(),
         }
     }
+
+    fn list_files(dir: PathBuf) -> Result<Vec<PathBuf>, io::Error>{
+        let mut output: Vec<String> = Vec::new();
+        if dir.is_dir(){
+
+        let  path: Vec<_> = fs::read_dir("dir").unwrap()
```

Comment:
Try to accomplish this with a combination of the following functions: `flatten()`, `map()`, `flat_map()`, and `collect()`.  This will let you handle everything recursively without using a `count` variable nor indexing into the vector.